Monika After Story/game/script-topics.rpy

```diff
@@ -15990,3 +15991,118 @@ label monika_stargazing:
     show monika 5kkbsa
     pause 2.0
     return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_player_eyesight",
+            category=["you"],
+            prompt="Wearing glasses"
+        )
+    )
+
+default persistent._mas_pm_wears_glasses = None
+default persistent._mas_pm_wears_lenses = None
+default persistent._mas_pm_poor_eyesight = None
+
+label monika_player_eyesight:
+    m 3lsb "Hey, [player], I was wondering about something lately."
+    m 1esa "Do you wear glasses, or maybe contact lenses?{nw}"
+    $ _history_list.pop()
+    menu:
+        m "Do you wear glasses, or maybe contact lenses?{fast}"
+
+        "I wear glasses.":
+            call monika_player_eyesight_awea(glasses=True)
+
+        "I wear contact lenses.":
+            call monika_player_eyesight_awea(glasses=False)
+
+        "I wear cosmetic glasses.":
+            call monika_player_eyesight_cg
+
+        "I wear them when needed.":
+            call monika_player_eyesight_eawn
+
+        "Nope, I don't.":
+            call monika_player_eyesight_n
+
+    return "derandom"
+
+# Always Wears Eyesight Aids
+label monika_player_eyesight_awea(glasses=False):
+    if glasses:
+        $ persistent._mas_pm_wears_glasses = True
+    else:
+        $ persistent._mas_pm_wears_lenses = True
+    $ persistent._mas_pm_poor_eyesight = True
+    m 2ekc "Oh, I see. "
+    extend 2eka "I really hope you see better with them on."
+    $ aids = "glasses" if glasses else "contact lenses"
+    m 1eka "Please don't neglect your eyesight, [player], as it can still deteriorate even if you constantly wear your [aids]."
+    m 1esa "If you've been on your computer for a while, maybe you could consider taking a little break to let your eyes rest."
+    m 3hsa "It'll take you just a few minutes a day, but I believe it would really help."
+    m 1ekc "Although... if it gets really bad, go see a doctor, okay?"
+    m 1eka "I really want my sweetheart to see this beautiful world just as clear as I do~"
+    return
+
+# Cosmetic Glasses
+label monika_player_eyesight_cg:
+    $ persistent._mas_pm_wears_glasses = False
+    $ persistent._mas_pm_wears_lenses = False
+    $ persistent._mas_pm_poor_eyesight = False
```

Comment:
You added `_mas_pm_poor_eyesight` which implies that you can wear glasses even if you don't have problems with your eyes. But at the same time you don't set `_mas_pm_wears_glasses` to `True` here. Same for `_mas_pm_wears_lenses`.

Reply:
`_mas_pm_poor_eyesight` is meant for the case when the player says that they wear their eyesight aids occasionally because of poor eyesight, which is reflected in the player model variable name.

If the player has no problem with their eyesight then all three of these variables should be `False`.